ffc/client.py

```diff
@@ -165,7 +188,139 @@ def _is_token_expired(self):
             return True
 
 
+class HttpxFFCAPIClient:
+    """
+    HTTPX API Client
+    """
+
+    def __init__(self, base_url: str, sub, secret):
+        self.api_base_url = base_url
+        self._sub = sub
+        self._secret = secret
+        self._jwt = None
+        self.client = httpx.AsyncClient(base_url=self.api_base_url)
+
+    async def get_generated_charge_files(self) -> list[dict[str, Any]] | None:
+        """
+        This method fetches the charges files in the GENERATED Status.
+        This status means that the charges files have been generated but
+        not yet processed by the billing procedure.
+        Raises:
+        """
+        try:
+            headers = self._get_headers()
+            rql_filter = "eq(status,generated)"
+            endpoint = f"{self.api_base_url}/ops/v1/charges?{rql_filter}"
+            response = await self.client.get(headers=headers, url=endpoint)
+            response.raise_for_status()
+            data =  response.json()
+            return data.get("items", [])
+        except httpx.HTTPError as error:
+            logger.exception(f"Request to get charge files failed: {error}")
+            return None
+
+    async def get_charges_file_download_url(
+        self, charge_file: dict[str, Any], tmp_dir: str
```

Comment:
Please rename `tmp_dir` to `download_dir` or something like that. The client doesn't care about the fact that the directory is temporary right ?